### PR TITLE
Fix rare edge case with extremely inbalanced data

### PR DESCRIPTION
**Pull request by mfeurer**
For dataset 360112 Auto-sklearn would fail because the data would
first be sub-sampled and then contain some classes only once.
In the internal splitting, the StratifiedShuffleSplit would not
be able to split the dataset into train and valid, and would resort
to only a ShuffleSplit. This could put the single sample for a
class into the test set. At predict time we would then miss one class.

This commit creates two new splitters which move a sample from the
test split to the training split if a class does not exist in the
train split.

### CODE REVIEW

**Comment by mfeurer** [comment]
CC @eddiebergman we could use these splitters to simplify `subsample_if_too_large`, right?

**Comment by eddiebergman** [comment]
> CC @eddiebergman we could use these splitters to simplify `subsample_if_too_large`, right?

Yes this splitter would probably work in that case too, or at least I see no reason it shouldn't.
